### DISCUSSION TITLE
GLES11Ext.GL_TEXTURE_EXTERNAL_OES unavaliable with black screen

### DISCUSSION THREAD

**Discussion by YuHayring** [Tencent/libpag#2442] [Q&A] [answered]
使用GL_TEXTURE_EXTERNAL_OES  播放黑屏 
版本 安卓 4.3.57

日志显示
tgfx:    A Graphic made from a texture can not be drawn on to a PAGSurface if its GPU context is not a share context to the PAGSurface.

`

        GLES20.glGenTextures(1, texture, 0)
        GLES20.glBindTexture(GLES11Ext.GL_TEXTURE_EXTERNAL_OES, texture[0])
        GLES20.glTexImage2D(GLES20.GL_TEXTURE_2D, 0, GLES20.GL_RGB, 256, 2, 0, GLES20.GL_RGB, GLES20.GL_UNSIGNED_BYTE, null)
        GLES20.glTexParameterf(GLES11Ext.GL_TEXTURE_EXTERNAL_OES, GL10.GL_TEXTURE_MIN_FILTER, GL10.GL_LINEAR.toFloat())
        GLES20.glTexParameterf(GLES11Ext.GL_TEXTURE_EXTERNAL_OES, GL10.GL_TEXTURE_MAG_FILTER, GL10.GL_LINEAR.toFloat())

        ...................

        
        pagFile.replaceImage(i, PAGImage.FromTexture(texture[0], GLES11Ext.GL_TEXTURE_EXTERNAL_OES, item.width, item.height))
`

**Comment by kevingpqi123** [comment, marked as answer]
这里的提示很明确哈，pag 内部渲染通过的是 OpenGL ES，自身会有一个 OpenGL 环境，你这边的场景应该是视频解码，解码出来的 OES 纹理自身也有一个 OpenGL 环境，需要通过 share context 来共享资源
